Argo/Functions/decode.swift

```diff
@@ -1,35 +1,215 @@
-// MARK: decode root objects and arrays
+/**
+  Decode a raw JSON value into an optional `Decodable` value
 
+  This function takes the output from `NSJSONSerialization` and attempts to
```

Comment:
would it be too confusing to say it takes `AnyObject` instead ... then later say "for example the output of NSJSONSerial..."?